diff --git a/node_modules/@lerna/conventional-commits/lib/constants.js b/node_modules/@lerna/conventional-commits/lib/constants.js
index 30d177d..25d2927 100644
--- a/node_modules/@lerna/conventional-commits/lib/constants.js
+++ b/node_modules/@lerna/conventional-commits/lib/constants.js
@@ -11,7 +11,7 @@ exports.COMMIT_GUIDELINE =
   "See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.";
 
 exports.CHANGELOG_HEADER = [
-  "# Change Log",
+  "# Changelog",
   "",
   "All notable changes to this project will be documented in this file.",
   exports.COMMIT_GUIDELINE,
